lib/internal/modules/esm/loader.js

```diff
@@ -1061,9 +1062,40 @@ function register(specifier, parentURL = undefined, options) {
   );
 }
 
+async function loadModule(specifier, parentURL, options = kEmptyObject) {
+  specifier = `${specifier}`;
+  parentURL = `${parentURL}`;
+
+  validateObject(options, 'options');
+  const { importAttributes = { __proto__: null } } = options;
+  validateObject(importAttributes, 'options.importAttributes');
+
+  const loader = getOrInitializeCascadedLoader();
+  const { url, format: resolvedFormat } = await loader.resolve(specifier, parentURL, importAttributes);
+  const result = await loader.load(url, { format: resolvedFormat, importAttributes });
+  const { source } = result;
+  let { format: finalFormat } = result;
+
+  // Translate internal formats to public ones.
+  if (finalFormat === 'commonjs-typescript') {
+    finalFormat = 'commonjs';
+  }
+  if (finalFormat === 'module-typescript') {
+    finalFormat = 'module';
+  }
```

Comment:
I think people may want to know it's typescript?

Also, this differs from hooks (which will receive the `*-typescript` formats. That could be confusing.